### PR TITLE
Make function handler checking to prevent main thread crash

### PR DESCRIPTION
**Pull request by AGritsevich**
Handler to observer can be set to null in any moment. Make check
before call handler.
Bug-fixes: [APPLINK-20011](https://adc.luxoft.com/jira/browse/APPLINK-20011)

Please review @anosach-luxoft, @malirod, @pestOO .

### CODE REVIEW

**Comment by AGritsevich** [comment]
@malirod, you right.

**Comment by anosach-luxoft** [comment]
@AGritsevich seems like we have duplicated logic on line 769 and line 794

**Comment by AGritsevich** [comment]
@malirod , I made it. Hask commit:968ac83..8304f61.
@anosach-luxoft, it diffrent commands. They even have diffrent input parameters.

**Comment by AGritsevich** [comment]
@malirod nowhere in code logging this case for this handler. So do and I.

**Comment by anosach-luxoft** [comment]
> @anosach-luxoft, it diffrent commands
What do you mean? What commands?
```
// In case of malformed message the connection should be broke up without
// any other notification to mobile.
if (close_reason != kMalformed && protocol_handler_) {
  protocol_handler_->SendEndSession(connection_handle, session_id);
}
```
```
// In case of malformed message the connection should be broke up without
// any other notification to mobile.
if (close_reason != kMalformed) {
  if (protocol_handler_) {
    protocol_handler_->SendEndSession(connection_handle, session_id);
  }
}
```

**Comment by AGritsevich** [comment]
@anosach-luxoft, I deleted copy past code.

**Comment by malirod** [comment]
Reviewed

**Comment by anosach-luxoft** [comment]
@AGritsevich AFAIK we have to call `OnServiceEndedCallback()` even in case `close_reason == kMalformed'. You can see how this logic realized in PASA. Please investigate this case.

**Comment by AGritsevich** [comment]
@anosach-luxoft, I move SendEndSesion like in PASA. What about another bad case?

**Comment by pestOO** [comment]
Update unittest, please.

**Comment by pestOO** [comment]
Reviewed

File: src/components/connection_handler/src/connection_handler_impl.cc [hunk 1]
```diff
@@ -767,7 +767,7 @@ void ConnectionHandlerImpl::CloseSession(ConnectionHandle connection_handle,
   // In case of malformed message the connection should be broke up without
   // any other notification to mobile.
   if (close_reason != kMalformed && protocol_handler_) {
-    protocol_handler_->SendEndSession(connection_handle, session_id);
+      protocol_handler_->SendEndSession(connection_handle, session_id);
```
Comment (anosach-luxoft) [inline src/components/connection_handler/src/connection_handler_impl.cc]: For what have you added these spaces?

Comment (AGritsevich) [inline src/components/connection_handler/src/connection_handler_impl.cc]: Those excess spaces. I've already deleted it.

**Pull request merged**